app/src/main/java/swati4star/createpdf/util/PDFUtils.java

```diff
@@ -654,42 +655,49 @@ private boolean isInputValid(String path, String[] ranges) {
      * ERROR_INVALID_INPUT  if input is invalid like -3 or 3--4 or 3,,4
      */
     public static int checkRangeValidity(int numOfPages, String[] ranges) {
-        int startPage;
-        int endPage;
-
+        int startPage = 0;
+        int endPage = 0;
+        int returnValue = NO_ERROR;
         if (ranges.length == 0)
-            return ERROR_INVALID_INPUT;
-
-        for (String range : ranges) {
-            if (!range.contains("-")) {
-                try {
-                    startPage = Integer.parseInt(range);
-                } catch (NumberFormatException e) {
-                    e.printStackTrace();
-                    return ERROR_INVALID_INPUT;
-                }
-                if (startPage > numOfPages || startPage == 0) {
-                    return ERROR_PAGE_NUMBER;
-                }
-            } else {
-                try {
-                    startPage = Integer.parseInt(range.substring(0, range.indexOf("-")));
-                    endPage = Integer.parseInt(range.substring(range.indexOf("-") + 1));
-                } catch (NumberFormatException e) {
-                    e.printStackTrace();
-                    return ERROR_INVALID_INPUT;
-                } catch (StringIndexOutOfBoundsException e) {
-                    e.printStackTrace();
-                    return ERROR_INVALID_INPUT;
-                }
-                if (startPage > numOfPages || endPage > numOfPages || startPage == 0 || endPage == 0) {
-                    return ERROR_PAGE_NUMBER;
-                } else if (startPage >= endPage) {
-                    return ERROR_PAGE_RANGE;
+            returnValue = ERROR_INVALID_INPUT;
+        else { 
+            for (String range : ranges) {
+                if (!range.contains("-")) {
```

Comment:
It'll be nice if you could add some comments on what exactly is being done here also